Math/NumberTheory/ArithmeticFunctions/Inverse.hs

```diff
@@ -347,16 +347,17 @@ instance Semiring MaxNatural where
 
 -- | Wrapper to use in conjunction with 'inverseTotient' and 'inverseSigma'.
 -- Extracts the minimal preimage of function.
--- 'Nothing' stands for a positive infinity.
-newtype MinNatural = MinNatural { unMinNatural :: Maybe Natural }
-  deriving (Show)
+data MinNatural
+  = MinNatural { unMinNatural :: !Natural }
```

Comment:
Right, I didn't give much importance to this before, but why was this `Maybe` before, and why change it now?

Reply:
It seems to me that switching to a newly defined sum type allows the non-infinite case to be more strict, since the indirection provided by `Just` is removed